packages/network/src/workers/syncUtils.ts

```diff
@@ -485,21 +491,58 @@ export function createTransformWorldEventsFromStream(decode: ReturnType<typeof c
   };
 }
 
+function mapTransactionHashesToNetworkComponentUpdates(events: NetworkComponentUpdate[]) {
```

Comment:
I do like explicit function names but feel like this one could be shorter (the input/output types are already described in the ts types so I don't think we need them in the function name too)
```suggestion
function groupByTxHash(events: NetworkComponentUpdate[]): { [key: string]: NetworkComponentUpdate[] } {
```